app/forms/developer_search_form.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class DeveloperSearchForm
+  include Virtus.model
+  include ActiveModel::Model
+
+  attribute :programming_language_id, Integer
+  attribute :language_id, Integer
+
+  def search
+    developers = Developer.all
+
+    developers = developers.joins(:developer_languages).where("developer_languages.language_id = #{language_id}") if language_id.present?
```

Comment:
Can you check the following page?
http://guides.rubyonrails.org/security.html#injection

Reply:
You should write more rigid code at the ActiveRecord level.
The `attribute` method is provided by Virtus. Who can say it never has bug? 

```ruby
# bad
Project.where("name = '#{params[:name]}'")
```
```ruby
# good
Project.where(name: params[:name])
or
Project.where("name = ?", params[:name])
```
```ruby
5.1.5@2.5.0 (main)> language_id = "1 OR 1 = 1"
=> "1 OR 1 = 1"
5.1.5@2.5.0 (main)> Developer.joins(:developer_languages).where("developer_languages.language_id = #{language_id}").count
   (0.6ms)  SELECT COUNT(*) FROM `developers` INNER JOIN `developer_languages` ON `developer_languages`.`developer_id` = `developers`.`id` WHERE (developer_languages.language_id = 1 OR 1 = 1)
=> 196
5.1.5@2.5.0 (main)> Developer.joins(:developer_languages).where(developer_languages: { language_id: language_id }).count
   (0.6ms)  SELECT COUNT(*) FROM `developers` INNER JOIN `developer_languages` ON `developer_languages`.`developer_id` = `developers`.`id` WHERE `developer_languages`.`language_id` = 1
=> 68
5.1.5@2.5.0 (main)> Developer.joins(:developer_languages).where("developer_languages.language_id = ?", language_id).count
   (0.6ms)  SELECT COUNT(*) FROM `developers` INNER JOIN `developer_languages` ON `developer_languages`.`developer_id` = `developers`.`id` WHERE (developer_languages.language_id = '1 OR 1 = 1')
=> 68
```